wasmplugin/plugin.go

```diff
@@ -231,44 +203,41 @@ func (ctx *httpContext) OnHttpRequestHeaders(numHeaders int, endOfStream bool) t
 	ctx.metrics.CountTX()
 
 	authority, err := proxywasm.GetHttpRequestHeader(":authority")
-	if err != nil {
-		return types.ActionContinue
-	}
+	if err == nil {
+		if waf, isDefault, resolveWAFErr := ctx.perAuthorityWAFs.getWAFOrDefault(authority); resolveWAFErr == nil {
+			ctx.tx = waf.NewTransaction()
 
-	wafDirectiveName, exist := ctx.perAuthorityWafSets.getdirectivesName(authority)
-	if !exist && ctx.wafDefaultEnabled {
-		ctx.tx, exist = ctx.wafSets.getTx(ctx.wafDefaultDirective)
-		if !exist {
-			return types.ActionContinue
-		}
+			logFields := []debuglog.ContextField{debuglog.Uint("context_id", uint(ctx.contextID))}
+			if !isDefault {
+				logFields = append(logFields, debuglog.Str("authority", authority))
+			}
+			ctx.logger = ctx.tx.DebugLogger().With(logFields...)
 
-		ctx.logger, exist = ctx.wafSets.getlogger(ctx.wafDefaultDirective)
-		if !exist {
-			return types.ActionContinue
-		}
-	} else {
-		ctx.tx, exist = ctx.wafSets.getTx(wafDirectiveName)
-		if !exist {
-			return types.ActionContinue
-		}
+			// CRS rules tend to expect Host even with HTTP/2
+			ctx.tx.AddRequestHeader("Host", authority)
+			ctx.tx.SetServerName(parseServerName(ctx.logger, authority))
 
-		ctx.logger, exist = ctx.wafSets.getlogger(wafDirectiveName)
-		if !exist {
+			if !isDefault {
+				ctx.metricLabelsKV = append(ctx.metricLabelsKV, "authority", authority)
```

Comment:
is this fixing bug metrics https://github.com/corazawaf/coraza-proxy-wasm/issues/186? Since we already append on ctx.metricLabelsKV, if the next request on the same stream (http2) is using a different authority, it's still a bug, right?

based on this docs, https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/proxywasm/types/context.go#LL100C2-L100C2
HttpContext is created on every HTTP stream

Reply:
I think it does as I am creating a new slice of metric labels on every new context.

Reply:
I think it does as I am creating a new slice of metric labels on every new context.

Reply:
thanks, just testing on my local, seems works